dhis-2/dhis-web/dhis-web-approval/src/main/java/org/hisp/dhis/approval/dataset/action/GenerateDataSetReportAction.java

```diff
@@ -251,13 +253,13 @@ public String execute()
 
         if ( formType.isCustom() && type == null )
         {
-            customDataEntryFormCode = dataSetReportService.getCustomDataSetReport( selectedDataSet, selectedPeriod,
-                selectedOrgunit, dimension, selectedUnitOnly );
+            customDataEntryFormCode = dataSetReportService.getCustomDataSetReport( selectedDataSet,
+                newArrayList( selectedPeriod ), selectedOrgunit, dimension, selectedUnitOnly );
```

Comment:
Why are these copied?

Reply:
Not copied, put from a single period into a list for compatibility with the changed service call. The newer WebAPI access point was upgraded to multiple periods, but the old Struts action was not.

Reply:
Oh, didn't see that. In that case it is better to use `Collections.singletonList` in the future.